ops/_main.py

```diff
@@ -552,9 +554,20 @@ def main(charm_class: Type[_charm.CharmBase], use_juju_for_storage: Optional[boo
 
     See `ops.main() <#ops-main-entry-point>`_ for details.
     """
-    try:
-        manager = _Manager(charm_class, use_juju_for_storage=use_juju_for_storage)
+    from . import tracing  # break circular import
 
-        manager.run()
-    except _Abort as e:
-        sys.exit(e.exit_code)
+    juju_context = _JujuContext.from_dict(os.environ)
+    tracing_manager = (
+        tracing.setup(juju_context, charm_class.__name__) if tracing else nullcontext()
+    )
```

Comment:
This is a follow up to a comment somewhere in here (sorry, it's a very large PR and I can't remember which of the "viewer" files I put it in) regarding `_JujuContext`. If this is the way it's used, then I think it's ok for `setup` to be global to the tracing module, but it should be private. That avoids charmers needing to know anything about `_JujuContext`, and it seems like we're doing the setup for them, so they don't need that method.

Reply:
So you're thinking:

- ops.tracing.Tracing
- ops.tracing.set_destination
- ops.tracing._setup
- ops.tracing._mark_observed